packages/cli/src/Server.ts

```diff
@@ -779,236 +798,259 @@ class App {
 		// Returns workflows
 		this.app.get(
 			`/${this.restEndpoint}/workflows`,
-			ResponseHelper.send(async (req: express.Request, res: express.Response) => {
-				const queryBuilder = Db.collections.Workflow!.createQueryBuilder('w');
-				queryBuilder.select(['w.id', 'w.name', 'w.active', 'w.createdAt', 'w.updatedAt']);
-				queryBuilder.leftJoinAndSelect('w.tags', 't');
-
-				if (req.query.filter) {
-					const jsonFilters = JSON.parse(req.query.filter as string);
-					const keys = Object.keys(jsonFilters);
-					keys.forEach((key) => {
-						queryBuilder.andWhere(`w.${key} = :${key}`, { [key]: jsonFilters[key] });
+			ResponseHelper.send(async (req: WorkflowRequest.GetAll) => {
+				let workflows: WorkflowEntity[] = [];
+
+				if (req.user.globalRole.name === 'owner') {
+					workflows = await Db.collections.Workflow!.find({
+						select: ['id', 'name', 'active', 'createdAt', 'updatedAt'],
+						relations: ['tags'],
+						where: req.query,
+					});
+				} else {
+					const shared = await Db.collections.SharedWorkflow!.find({
+						relations: ['workflow', 'workflow.tags'],
+						where: buildWhereClause({
+							user: req.user,
+							entityType: 'workflow',
+						}),
+					});
+
+					if (!shared.length) {
+						throw new ResponseHelper.ResponseError('No workflows found', undefined, 404);
+					}
+
+					workflows = await Db.collections.Workflow!.find({
+						relations: ['tags'],
+						select: ['id', 'name', 'active', 'createdAt', 'updatedAt'],
+						where: {
+							id: In(shared.map(({ workflow }) => workflow.id)),
+							...req.query,
+						},
 					});
 				}
-				// Simplified checks for now. Get only
-				// workflows I have access to.
-				queryBuilder.innerJoin('w.shared', 'shared');
-				queryBuilder.andWhere('shared.userId = :userId', { userId: (req.user as User).id });
 
-				const workflows = await queryBuilder.getMany();
+				return workflows.map((workflow) => {
+					const { id, tags, ...rest } = workflow;
 
-				workflows.forEach((workflow) => {
-					// @ts-ignore
-					workflow.id = workflow.id.toString();
-					// @ts-ignore
-					workflow.tags = workflow.tags.map(({ id, name }) => ({ id: id.toString(), name }));
+					return {
+						id: id.toString(),
+						...rest,
+						tags: tags?.map(({ id, ...rest }) => ({ id: id.toString(), ...rest })) ?? [],
+					};
```

Comment:
Copying instead of mutating here is inefficient, but TS is right to complain about forcing the field type change without adjusting the type in the model, and the model's field type should not be adjusted - it should reflect the column type.

Hence I think we need to have workflow ID and tag ID types in the FE match exactly our class models to stop doing this needless copy or mutation work - this applies to other workflow endpoints as well.